libopenage/pathfinding/cost_field.h

```diff
@@ -95,15 +103,33 @@ class CostField {
 	 * Set the cost field values.
 	 *
 	 * @param cells Cost field values.
+	 * @param changed Time at which the cost value is changed.
+	 */
+	void set_costs(std::vector<cost_t> &&cells, const time::time_t &changed);
+
+	/**
+	 * Check if the cost field is dirty at the specified time.
+	 *
+	 * @param time Cost field is dirty if the cost field is accessed after the time given in valid_until.
```

Comment:
You should always explain what `time` represents/is used for. From your docstring, it's not apparent what a user of this method should pass for `time`.

```suggestion
	 * @param time Time of access to the cost field.
```